clients/v1_repl/autocomplete.go

```diff
@@ -0,0 +1,211 @@
+				// remainder is everything after the just-matched term
+				rem := strings.TrimLeft(remainder[len(term):], " ")
+				// if sugsuggestFn is nil, this is a terminating suggestion (leaf node with no more following suggestions)
+				if node.subsuggestFn == nil {
+					return []prompt.Suggest{}, rem
+				}
+				sugs, rem := node.subsuggestFn(rem)
+				return getSuggestions(rem, sugs)
+			}
+		}
+	}
+	// if no match was found, convert the s map into suggestions and filter by the remaining characters
+	var sugs []prompt.Suggest
+	for text, node := range s {
+		sugs = append(sugs, prompt.Suggest{Text: text, Description: node.Description})
+	}
+	return prompt.FilterFuzzy(sugs, remainder, true), remainder
+}
+
+func (c *Client) completer(d prompt.Document) []prompt.Suggest {
+	// the commented-out lines are unsupported in 2.x
+	suggestions := map[string]SuggestNode{
+		"use":    {Description: "Set current database", subsuggestFn: c.suggestUse},
+		"pretty": {Description: "Toggle pretty print for the json format"},
+		"precision": {Description: "Specify the format of the timestamp", subsuggestFn: func(rem string) (map[string]SuggestNode, string) {
+			return map[string]SuggestNode{
+				"rfc3339": {},
+				"ns":      {},
+				"u":       {},
+				"ms":      {},
+				"s":       {},
+				"m":       {},
+				"h":       {},
+			}, rem
+		}},
+		"history":  {Description: "Display shell history"},
+		"settings": {Description: "Output the current shell settings"},
+		"clear": {Description: "Clears settings such as database", subsuggestFn: func(rem string) (map[string]SuggestNode, string) {
+			return map[string]SuggestNode{
+				"db":               {},
+				"database":         {},
+				"retention policy": {},
+				"rp":               {},
+			}, rem
+		}},
+		"exit":   {Description: "Exit the InfluxQL shell"},
+		"quit":   {Description: "Exit the InfluxQL shell"},
+		"gopher": {Description: "Display the Go Gopher"},
+		"help":   {Description: "Display help options"},
+		"format": {Description: "Specify the data display format", subsuggestFn: func(rem string) (map[string]SuggestNode, string) {
+			return map[string]SuggestNode{
+				"column": {},
+				"csv":    {},
+				"json":   {},
+			}, rem
+		}},
+		"SELECT":      {subsuggestFn: c.suggestSelect},
+		"INSERT":      {},
+		"INSERT INTO": {},
+		"DELETE":      {subsuggestFn: c.suggestDelete},
+		"SHOW": {subsuggestFn: func(rem string) (map[string]SuggestNode, string) {
+			return map[string]SuggestNode{
+				// "CONTINUOUS QUERIES":            {},
+				"DATABASES": {},
+				// "DIAGNOSTICS":                   {},
+				"FIELD KEY CARDINALITY": {},
+				"FIELD KEYS":            {},
+				// "GRANTS":                {},
+				// "MEASUREMENT CARDINALITY":       {},
+				"MEASUREMENT EXACT CARDINALITY": {},
+				"MEASUREMENTS":                  {},
+				// "QUERIES":                 {},
+				// "RETENTION POLICIES":      {},
+				"SERIES": {},
+				// "SERIES CARDINALITY": {},
+				"SERIES EXACT CARDINALITY": {},
```

Comment:
Does this work against 2.x / cloud?

Reply:
Yeah, at least for 2.x OSS, `SHOW SERIES EXACT CARDINALITY` is implemented while `SHOW SERIES CARDINALITY` is not.